### PR TITLE
Added DEBUG switch, scrollbar_size property, fixed text_log line length

### PR DESCRIPTION
**Pull request by willmcgugan**
Fixed a few issues while looking in to something else.

- Added a `TEXTUAL_DEBUG` constant set from an env var, that checks for a common logic error.
- Added a `Strip.apply_style` which caches the moderately expensive operation.
- Fixes a miscalculation in TextLog

### CODE REVIEW

File: src/textual/strip.py [hunk 1]
```diff
@@ -39,8 +54,13 @@ def __init__(
         self._cell_length = cell_length
         self._divide_cache: FIFOCache[tuple[int, ...], list[Strip]] = FIFOCache(4)
         self._crop_cache: FIFOCache[tuple[int, int], Strip] = FIFOCache(4)
+        self._style_cache: FIFOCache[Style, Strip] = FIFOCache(4)
```
Comment (davep) [inline src/textual/strip.py:57]: Is there something in the context that makes it clear what `4` is for?

Comment (willmcgugan) [reply src/textual/strip.py:57]: It's fairly arbitrary, predicated on observation that there are relatively few combinations of inputs.

File: src/textual/widget.py [hunk 2]
```diff
@@ -1029,6 +1029,11 @@ def scrollbars_enabled(self) -> tuple[bool, bool]:
         enabled = self.show_vertical_scrollbar, self.show_horizontal_scrollbar
         return enabled
 
+    @property
+    def scrollbar_size(self) -> Size:
+        """The dimensions of the scrollbar stored in a Size object."""
```
Comment (davep) [inline src/textual/widget.py:1034]: I'm finding this docstring a wee bit confusing; it makes a `Size` from `scrollbar_size_vertical` and `scrollbar_size_horizontal`, but they're two *different* scrollbars, as I understand it. Perhaps the docstring could do with being expanded to make it clear that it's about two different scrollbars and also make it clear which is in which `Size` attribute?

**Review by davep — APPROVED** [review]
Small docstring thing but otherwise all good with me!

File: src/textual/constants.py [hunk 3]
```diff
@@ -3,9 +3,32 @@
 
 """
 
+"""
+A number of app level constants where the value is pulled from environment variables.
+"""
```
Comment (rodrigogiraoserrao) [inline src/textual/constants.py:8]: I don't understand why we have two module-level docstrings here! 😕

File: src/textual/strip.py [hunk 4]
```diff
@@ -10,9 +10,23 @@
...
+def get_line_length(segments: Iterable[Segment]) -> int:
+    """Get the line length (total length of all segments).
+
+    Args:
+        segments: Iterable of segments.
+
+    Returns:
+        Length of line in cells.
+    """
+    _cell_len = cell_len
```
Comment (rodrigogiraoserrao) [inline src/textual/strip.py:26]: Is this a micro-optimisation?

**Review by rodrigogiraoserrao — APPROVED** [review]

File: src/textual/widget.py (hunk 2 above)
Comment (willmcgugan) [reply src/textual/widget.py:1034]: It's poorly named, and could just be a tuple. I will clarify.

File: src/textual/strip.py (hunk 4 above)
Comment (willmcgugan) [reply src/textual/strip.py:26]: It is. Globals are slower than locals.

**Pull request merged**